sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -0,0 +1,232 @@
+
+public class ScannerFileServiceTest {
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test
+  public void isSupportedAbsolute_passes_correct_argument() {
+    // arrange
+    FileSystem fs = mock(FileSystem.class);
+    FilePredicates filePredicates = mock(FilePredicates.class);
+
+    ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+    when(filePredicates.hasAbsolutePath(argumentCaptor.capture())).thenReturn(mock(FilePredicate.class));
+    when(fs.predicates()).thenReturn(filePredicates);
+    when(fs.hasFiles(any())).thenReturn(true);
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    sut.isSupportedAbsolute("/_/some/path/file.cs");
+
+    // assert
+    assertThat(argumentCaptor.getValue()).isEqualTo("/_/some/path/file.cs");
+    assertThat(logTester.logs()).isEmpty();
+  }
+
+  @Test
+  public void isSupportedAbsolute_returns_fileSystem_result_when_true() {
+    FileSystem fs = createFileSystemForHasFiles(true);
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    boolean result = sut.isSupportedAbsolute("/_/some/path/file.cs");
+
+    assertThat(result).isTrue();
+    assertThat(logTester.logs()).isEmpty();
+  }
+
+  @Test
+  public void isSupportedAbsolute_returns_fileSystem_result_when_false() {
+    FileSystem fs = createFileSystemForHasFiles(false);
+
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    boolean result = sut.isSupportedAbsolute("/_/some/path/file.cs");
+
+    assertThat(result).isFalse();
+    assertThat(logTester.logs()).isEmpty();
+  }
+
+  @Test
+  public void getFilesByRelativePath_retrieves_all_files() {
+    // arrange
+    FileSystem fs = mock(FileSystem.class);
+    FilePredicates filePredicates = mock(FilePredicates.class);
+    FilePredicate allMock = mock(FilePredicate.class);
+    FilePredicate languageKeyMock = mock(FilePredicate.class);
+    FilePredicate andMock = mock(FilePredicate.class);
+
+    ArgumentCaptor<FilePredicate> argumentCaptor = ArgumentCaptor.forClass(FilePredicate.class);
+
+    when(fs.inputFiles(argumentCaptor.capture())).thenReturn(Collections.emptyList());
+
+    when(filePredicates.hasLanguage("key")).thenReturn(languageKeyMock);
+    when(filePredicates.all()).thenReturn(allMock);
+    when(filePredicates.and(allMock, languageKeyMock)).thenReturn(andMock);
+    when(fs.predicates()).thenReturn(filePredicates);
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    sut.getFilesByRelativePath("foo");
+
+    // assert
+    assertThat(argumentCaptor.getValue()).isEqualTo(andMock);
+  }
+
+  @Test
+  public void getFilesByRelativePath_when_no_indexed_files_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Collections.emptyList());
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
+    Optional<InputFile> result = sut.getFilesByRelativePath("C:\\_\\some\\path\\file.cs");
```

Comment:
Have you seen the `C:\_\` path on some generated coverage file? 
Do we know what happens when you build from `\\server\path\` ?

Reply:
No, I've only seen something like `/_/x/y/Foo.cs`, including on Windows systems

but when the input `path` is `/_/its/projects/CoverageWithDeterministicSourcePaths/CoverageWithDeterministicSourcePaths/Foo.cs`, [new File(path).getCanonicalPath()](https://github.com/SonarSource/sonar-dotnet/blob/6ef1c842c09232c065b477f79910effb8554523f/sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioCoverageXmlReportParser.java#L111) will return `C:\_\its\projects\CoverageWithDeterministicSourcePaths\CoverageWithDeterministicSourcePaths\Foo.cs` 

So the drive letter is more for the behavior of the Java library call that we use.

Reply:
I've mentioned this behavior in [this comment](https://github.com/SonarSource/sonar-dotnet/issues/3362#issuecomment-738183321) as well

Reply:
I now see the logic is twisted:
- the deterministic source paths are unix style `/_/x/y/Foo.cs`
- then `getCanonicalPath` makes them Windows style on Windows systems
- then we transform it again to unix style to match the scanner normalized paths

From this point of view, it's better to work with the original path from the beginning instead of the canonicalPath, when the canonicalPath is not found as an absolute path by the scanner.